clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1272,6 +1272,8 @@ private CompletedFetch initializeCompletedFetch(CompletedFetch nextCompletedFetc
                 log.debug("Received unknown leader epoch error in fetch for partition {}", tp);
             } else if (error == Errors.UNKNOWN_SERVER_ERROR) {
                 log.warn("Unknown error fetching data for topic-partition {}", tp);
+            } else if (error == Errors.CORRUPT_MESSAGE) {
+                log.warn("Encountered corrupt message while fetching data for topic-partition {}", tp);
```

Comment:
I think the main question here is whether we want to raise this error to the user. If we happen to see message corruption in the fetched data, we raise that to the user, so should this case be different?

Reply:
Ack. As discussed, will (1) change this to correspond to the current behavior for corruption in fetched data, i.e. throwing a `KafkaException` that wraps around a `CorruptRecordException`, and (2) remove the log message expectation.

Reply:
Correction: _just_ a `KafkaException` (the wrapping around I referred to was the _text_ of the `CorruptRecordException`, not the exception itself)